### DISCUSSION TITLE
Blank page after fresh manual install

### DISCUSSION THREAD

**Discussion by patrikroy** [JulianPrieber/littlelink-custom#510] [Questions] [answered]
Hi,
I tried to get LinkStack working on my self-hosted server, but nothing's showing up. Here is my setup and process :

`Ubuntu 22.04` / `Apache 2.4.52` / `PHP 8.1.2` / `MySQL 8.0.33-0` / `sqlite 3.37.2`
On Apache is enabled `.htaccess` support, `mod_rewrite`, `https` on the Virtual Host. The [required php modules](https://docs.linkstack.org/getting-started/requirements/) are installed.

I got the project file through a `wget https://github.com/LinkStackOrg/LinkStack/archive/refs/heads/main.zip` and stored the files in `/var/www/mywebsite.com/linkstack/`

Here are the files and rights :
```
drwxrwxr-x  9 patrikroy patrikroy   4096 juil. 27 16:04 app
-rwxr-xr-x  1 patrikroy patrikroy   1686 juil. 27 16:04 artisan
drwxrwxr-x 16 patrikroy patrikroy   4096 juil. 27 16:04 assets
drwxrwxr-x  3 patrikroy patrikroy   4096 juil. 27 16:04 bootstrap
-rw-rw-r--  1 patrikroy patrikroy   5278 juil. 27 16:04 CODE_OF_CONDUCT.md
-rw-rw-r--  1 patrikroy patrikroy   2200 juil. 27 16:04 composer.json
-rw-rw-r--  1 patrikroy patrikroy 391687 juil. 27 16:04 composer.lock
drwxrwxr-x  2 patrikroy patrikroy   4096 juil. 28 08:15 config
-rw-rw-r--  1 patrikroy patrikroy   2736 juil. 27 16:04 CONTRIBUTING.md
drwxrwxr-x  5 patrikroy patrikroy   4096 juil. 27 16:04 database
-rwxr-xr-x  1 patrikroy patrikroy    220 juil. 27 16:04 .editorconfig
-rw-rw-r--  1 patrikroy patrikroy   4876 juil. 27 16:04 .env
-rwxr-xr-x  1 patrikroy patrikroy    811 juil. 27 16:04 .env.example
-rwxr-xr-x  1 patrikroy patrikroy    111 juil. 27 16:04 .gitattributes
drwxrwxr-x  3 patrikroy patrikroy   4096 juil. 27 16:04 .github
-rwxr-xr-x  1 patrikroy patrikroy    220 juil. 27 16:04 .gitignore
-rw-rw-r--  1 patrikroy patrikroy    840 juil. 27 16:04 .htaccess
-rw-rw-r--  1 patrikroy patrikroy   2377 juil. 27 16:04 index.php
-rw-rw-r--  1 patrikroy patrikroy      0 juil. 27 16:04 INSTALLING
-rw-rw-r--  1 patrikroy patrikroy  34523 juil. 27 16:04 LICENSE
-rw-rw-r--  1 patrikroy patrikroy     71 juil. 27 16:04 mix-manifest.json
-rw-rw-r--  1 patrikroy patrikroy    648 juil. 27 16:04 package.json
-rw-rw-r--  1 patrikroy patrikroy 748978 juil. 27 16:04 package-lock.json
-rwxr-xr-x  1 patrikroy patrikroy   1202 juil. 27 16:04 phpunit.xml
-rw-rw-r--  1 patrikroy patrikroy  14193 juil. 27 16:04 README.md
drwxrwxr-x  7 patrikroy patrikroy   4096 juil. 27 16:04 resources
drwxrwxr-x  2 patrikroy patrikroy   4096 juil. 27 16:04 routes
-rw-rw-r--  1 patrikroy patrikroy    844 juil. 27 16:04 SECURITY.md
-rwxr-xr-x  1 patrikroy patrikroy    545 juil. 27 16:04 server.php
drwxrwxr-x  7 patrikroy patrikroy   4096 juil. 27 16:04 storage
-rwxr-xr-x  1 patrikroy patrikroy    181 juil. 27 16:04 .styleci.yml
drwxrwxr-x  4 patrikroy patrikroy   4096 juil. 27 16:04 themes
-rw-rw-r--  1 patrikroy patrikroy      6 juil. 27 16:04 version.json
-rw-rw-r--  1 patrikroy patrikroy   1183 juil. 27 16:04 web.config
-rwxr-xr-x  1 patrikroy patrikroy    613 juil. 27 16:04 webpack.mix.js
```

When calling `https://mywebsite.com/linkstack/` I get a blank page with no code in it.

What am I doing wrong, any clues?

**Comment by JulianPrieber** [comment]
You need to download the latest release

**Reply by patrikroy** [reply, marked as answer]
Ok thanks for your quick answer, I thought the .zip was the lastest.

Just got the 4.2.2, but I still have error message :
`The stream or file "/var/www/_[redacted]_/linkstack/storage/logs/laravel.log" could not be opened in append mode: Failed to open stream: Permission denied The exception occurred while attempting to log: The stream or file "/var/www/_[redacted]_/linkstack/storage/logs/laravel.log" could not be opened in append mode: Failed to open stream: Permission denied The exception occurred while attempting to log: copy(/var/www/_[redacted]_/linkstack/config/advanced-config.php): Failed to open stream: Permission denied Context: {"exception":{}} Context: {"exception":{}}`

I'll try to check the permissions.

[UPDATE]
Changed owner/group of linkstack files and directories to `www-data`
It then asked me to generate an application key with `php artisan key:generate`, did that
Now stuck to another challenge:

> Illuminate \ Database \ QueryException
> PHP 8.1.2-1ubuntu2.13
> 9.52.10
> could not find driver
> PRAGMA foreign_keys = ON;

[UPDATE2]
Problem was I got `sqlite3` installed, which was not the same as `php-sqlite3`
Did `sudo apt-get install php-sqlite3` and was able to reload the LinkStack setup.

Hope it will help others, thanks @JulianPrieber
